### PR TITLE
Improve UX when we can't install a package

### PR DESCRIPTION
**Pull request by FabioRosado**
Currently when we try to install a package that does not exist or there are no pure wheels for, things fail *silently* (we do have console logs), and the splash screen also doesn't close.

This PR handles exceptions on `await micropip.install` and throws a new error InstallError (should we reuse the fetch error instead?). It also addresses a comment left by @antocuni  that, ideally, we should only load micropip if we have packages to install.

Screenshot from 2022-11-30 18-41-43

After this PR, the load screen will close and the alert banner will be shown with a helpful message. Unfortunately, we can't simply use the exception message because it contains a big Python traceback.

Fixes: #911

### CODE REVIEW

File: pyscriptjs/src/pyodide.ts [hunk 1]
```diff
@@ -89,8 +92,36 @@ export class PyodideRuntime extends Runtime {
         if (package_name.length > 0) {
             logger.info(`micropip install ${package_name.toString()}`);
             const micropip = this.globals.get('micropip') as Micropip;
-            await micropip.install(package_name);
-            micropip.destroy();
+            try{
```
Comment (madhur-tandon) [inline pyscriptjs/src/pyodide.ts:95]: nitpick but `try{` --> `try {`

File: pyscriptjs/tests/integration/test_01_basic.py [hunk 2]
```diff
@@ -101,6 +101,47 @@ def test_packages(self):
...
+        )
+
+        alert_banner = self.page.wait_for_selector(".alert-banner")
+        assert expected_alert_banner_msg in alert_banner.inner_text()
+
+    def test_no_python_wheel(self):
+        self.pyscript_run(
+            """
+            <py-config>
+                # we use asciitree because it's one of the smallest packages
```
Comment (madhur-tandon) [inline pyscriptjs/tests/integration/test_01_basic.py:129]: same here, comment shouldn't be present

**Review by madhur-tandon — APPROVED** [review]
Minor suggestions but looks good!
Do we want to perhaps show the traceback in a clickable dropdown? Just a suggestion.

**Comment by FabioRosado** [comment]
Thanks just pushed the changes and also added docs for the error in the console can you check if it looks good?

Yeah showing the traceback would be nice, but I think its unlikely we get many python tracebacks in this banner 🤔

File: pyscriptjs/src/pyodide.ts [hunk 3]
```diff
@@ -89,8 +92,38 @@ export class PyodideRuntime extends Runtime {
         if (package_name.length > 0) {
             logger.info(`micropip install ${package_name.toString()}`);
             const micropip = this.globals.get('micropip') as Micropip;
-            await micropip.install(package_name);
-            micropip.destroy();
+            try {
+                await micropip.install(package_name);
+                micropip.destroy();
+            } catch(e) {
```
Comment (antocuni) [inline pyscriptjs/src/pyodide.ts:98]: here we should check that `e` is actually a Python error and not some unrelated exception that we don't expect

**Review by antocuni — APPROVED** [review]
LGTM, I like it :).
Also, I think that this should also speed up the loading time of all the pages which don't use `config.packages`. I wonder by how many

File: pyscriptjs/src/pyodide.ts (hunk 3 above)
Comment (FabioRosado) [reply pyscriptjs/src/pyodide.ts:98]: In that case we are adding the exception message to the banner. Would that be enough you think?
If we handle only `PythonError` exceptions, the others will only be visible in the console logs and the splashscreen wont be closed 🤔

Comment (antocuni) [reply pyscriptjs/src/pyodide.ts:98]: > In that case we are adding the exception message to the banner. Would that be enough you think? 

I suppose we can live with that, but it's conceptually wrong, see below.

> If we handle only `PythonError` exceptions, the others will only be visible in the console logs and the splashscreen wont be closed thinking

yes, but this is a separate problem: generally speaking, Python errors are caused by the user, but JS errors are caused by us: ideally, we should catch them and display a banner to the user which says "pyscript internal error, please open an issue": this way, they know that it's not *their* fault.

**Pull request merged**